pkg/controller/bastion/actuator.go

```diff
@@ -184,19 +184,26 @@ func getPublicIP(ctx context.Context, log logr.Logger, factory azureclient.Facto
 	return ip, nil
 }
 
-func getSubnet(ctx context.Context, log logr.Logger, factory azureclient.Factory, vNet, subnetWork string, opt *Options) (*network.Subnet, error) {
+func getSubnet(ctx context.Context, log logr.Logger, factory azureclient.Factory, infrastructureStatus *azure.InfrastructureStatus, opt *Options) (*network.Subnet, error) {
+	var sg string
 	subnetClient, err := factory.Subnet(ctx, opt.SecretReference)
 	if err != nil {
 		return nil, err
 	}
 
-	subnet, err := subnetClient.Get(ctx, opt.ResourceGroupName, vNet, subnetWork, "")
+	if infrastructureStatus.Networks.VNet.ResourceGroup != nil {
+		sg = *infrastructureStatus.Networks.VNet.ResourceGroup
+	} else {
+		sg = opt.ResourceGroupName
+	}
+
+	subnet, err := subnetClient.Get(ctx, sg, infrastructureStatus.Networks.VNet.Name, infrastructureStatus.Networks.Subnets[0].Name, "")
```

Comment:
sorry for the late comment, I only got now aware of this PR as the corresponding issue was closed. Shouldn't the length of infrastructureStatus.Networks.Subnets be checked before? Or is it guaranteed at this point that it's not empty?

Reply:
It is done here: https://github.com/tedteng/gardener-extension-provider-azure/blob/bdd779563772700d8883ace9ceb911f121f52817/pkg/controller/bastion/actuator_reconcile.go#L145

Reply:
thanks @kon-angelo